libraries/NonFungibleToken/test/nonFungibleToken-tests.js

```diff
@@ -0,0 +1,368 @@
+					const ownerPublicKey = crypto.aeEncodeKey(bytes.toBytes(encodedData.value, true))
+
+					const decodedBalanceOfResult = await balanceOfResult.decode("int");
+
+					assert.equal(ownerPublicKey, config.ownerKeyPair.publicKey)
+					assert.equal(decodedBalanceOfResult.value, expectedBalance)
+				})
+
+				it('should not mint from non-owner', async () => {
+					const unauthorisedPromise = secondClient.contractCall(compiledContract.bytecode, 'sophia', deployedContract.address, "mint", {
+						args: `(${firstTokenId}, ${utils.publicKeyToHex(config.ownerKeyPair.publicKey)})`,
+						options: {
+							ttl: config.ttl
+						}
+					})
+					await assert.isRejected(unauthorisedPromise, errorMessages.ONLY_OWNER_CAN_MINT);
+				})
+
+				it('should not mint token with id that already exist', async () => {
+					//Arrange
+
+					//Act
+					const secondDeployContractPromise = deployedContract.call('mint', {
+						args: `(${firstTokenId}, ${utils.publicKeyToHex(config.ownerKeyPair.publicKey)})`,
+						options: {
+							ttl: config.ttl
+						}
+					})
+
+					//Assert
+					await assert.isRejected(secondDeployContractPromise, errorMessages.CANNOT_OVERRIDE_TOKEN);
+				})
+			})
+
+			describe('Burn', () => {
+				it('should burn token successfully', async () => {
+					//Arrange
+					const expectedBalance = 0;
+
+					//Act
+					const ownerOfPromise = deployedContract.call('burn', {
+						args: `(${firstTokenId})`,
+						options: {
+							ttl: config.ttl
+						}
+					});
+					
+					await ownerOfPromise;
+
+					const balanceOfPromise = deployedContract.call('balanceOf', {
+						args: `(${utils.publicKeyToHex(config.ownerKeyPair.publicKey)})`,
+						options: {
+							ttl: config.ttl
+						}
+					});
+					
+					const balanceOfResult = await balanceOfPromise;
+
+					//Assert
+					const decodedBalanceOfResult = await balanceOfResult.decode("int");
+					assert.equal(decodedBalanceOfResult.value, expectedBalance)
+				})
+
+				it('shouldn`t burn token from non-owner', async () => {
+					//Arrange
+
+					//Act
+					const unauthorizedBurnPromise = secondClient.contractCall(compiledContract.bytecode, 'sophia', deployedContract.address, "burn", {
+						args: `(${firstTokenId})`,
+						options: {
+							ttl: config.ttl
+						}
+					})
+
+					//Assert
+					await assert.isRejected(unauthorizedBurnPromise, errorMessages.ONLY_OWNER_CAN_TRANSFER);
+				})
+			})
+
+			describe('Transfer', () => {
+				it('should transfer token successfully', async () => {
+					//Arrange
+					const expectedBalanceOfNotOwner = 1;
+					const expectedBalanceOfOwner = 0;
+
+					//Act
+					const setApprovalForAllPromise = deployedContract.call('setApprovalForAll', {
+						args: `(${utils.publicKeyToHex(config.ownerKeyPair.publicKey)},${true})`,
+						options: {
+							ttl: config.ttl
+						}
+					});
+					
+					await setApprovalForAllPromise;
+
+					const approvePromise = deployedContract.call('approve', {
```

Comment:
Not sure if should be addressed by this PR, but I feel that it might prove useful if we take the methods names to a object like 
```
const NFTMethods = {
APPROVE: 'approve',
TRANSFER: 'transfer'
}
```

and us e it here like 

deployedContract.call(NFTMethods.APPROVE,...). The object will act somewhat as a definition of the methods of the contract.

What do you think?